PR: https://github.com/odoo/odoo/pull/

From: d4bae06d081798e35529c2323b66ec5467545cdb
From: Christophe Matthieu
Date: 2015-08-21 20:06:52

Structural Changes: 3
Total Changes: 179

[IMP] web: use css to display attachment mimetype

Widgets and website display images or display the type with icon like pdf, doc...

================================= pseudo patch: =================================

--- a/addons/mail/models/__init__.py
+++ b/addons/mail/models/__init__.py
@@ -1,6 +1,5 @@
 # -*- coding: utf-8 -*-
 
-import ir_attachment
 import mail_message_subtype
 import mail_tracking_value
 import mail_alias

--- a/addons/mail/models/ir_attachment.py
+++ b/None
@@ -1,174 +0,0 @@
-# -*- coding: utf-8 -*-
-
-import os.path
-
-from openerp import api, fields, models
-
-
-class IrAttachment(models.Model):
-    """ Update partner to add a field about notification preferences """
-    _name = "ir.attachment"
-    _inherit = 'ir.attachment'
-
-    _fileext_to_type = {
-        '7z': 'archive',
-        'aac': 'audio',
-        'ace': 'archive',
-        'ai': 'vector',
-        'aiff': 'audio',
-        'apk': 'archive',
-        'app': 'binary',
-        'as': 'script',
-        'asf': 'video',
-        'ass': 'text',
-        'avi': 'video',
-        'bat': 'script',
-        'bin': 'binary',
-        'bmp': 'image',
-        'bzip2': 'archive',
-        'c': 'script',
-        'cab': 'archive',
-        'cc': 'script',
-        'ccd': 'disk',
-        'cdi': 'disk',
-        'cdr': 'vector',
-        'cer': 'certificate',
-        'cgm': 'vector',
-        'cmd': 'script',
-        'coffee': 'script',
-        'com': 'binary',
-        'cpp': 'script',
-        'crl': 'certificate',
-        'crt': 'certificate',
-        'cs': 'script',
-        'csr': 'certificate',
-        'css': 'html',
-        'csv': 'spreadsheet',
-        'cue': 'disk',
-        'd': 'script',
-        'dds': 'image',
-        'deb': 'archive',
-        'der': 'certificate',
-        'djvu': 'image',
-        'dmg': 'archive',
-        'dng': 'image',
-        'doc': 'document',
-        'docx': 'document',
-        'dvi': 'print',
-        'eot': 'font',
-        'eps': 'vector',
-        'exe': 'binary',
-        'exr': 'image',
-        'flac': 'audio',
-        'flv': 'video',
-        'gif': 'webimage',
-        'gz': 'archive',
-        'gzip': 'archive',
-        'h': 'script',
-        'htm': 'html',
-        'html': 'html',
-        'ico': 'image',
-        'icon': 'image',
-        'img': 'disk',
-        'iso': 'disk',
-        'jar': 'archive',
-        'java': 'script',
-        'jp2': 'image',
-        'jpe': 'webimage',
-        'jpeg': 'webimage',
-        'jpg': 'webimage',
-        'jpx': 'image',
-        'js': 'script',
-        'key': 'presentation',
-        'keynote': 'presentation',
-        'lisp': 'script',
-        'lz': 'archive',
-        'lzip': 'archive',
-        'm': 'script',
-        'm4a': 'audio',
-        'm4v': 'video',
-        'mds': 'disk',
-        'mdx': 'disk',
-        'mid': 'audio',
-        'midi': 'audio',
-        'mkv': 'video',
-        'mng': 'image',
-        'mp2': 'audio',
-        'mp3': 'audio',
-        'mp4': 'video',
-        'mpe': 'video',
-        'mpeg': 'video',
-        'mpg': 'video',
-        'nrg': 'disk',
-        'numbers': 'spreadsheet',
-        'odg': 'vector',
-        'odm': 'document',
-        'odp': 'presentation',
-        'ods': 'spreadsheet',
-        'odt': 'document',
-        'ogg': 'audio',
-        'ogm': 'video',
-        'otf': 'font',
-        'p12': 'certificate',
-        'pak': 'archive',
-        'pbm': 'image',
-        'pdf': 'print',
-        'pem': 'certificate',
-        'pfx': 'certificate',
-        'pgf': 'image',
-        'pgm': 'image',
-        'pk3': 'archive',
-        'pk4': 'archive',
-        'pl': 'script',
-        'png': 'webimage',
-        'pnm': 'image',
-        'ppm': 'image',
-        'pps': 'presentation',
-        'ppt': 'presentation',
-        'ps': 'print',
-        'psd': 'image',
-        'psp': 'image',
-        'py': 'script',
-        'r': 'script',
-        'ra': 'audio',
-        'rar': 'archive',
-        'rb': 'script',
-        'rpm': 'archive',
-        'rtf': 'text',
-        'sh': 'script',
-        'sub': 'disk',
-        'svg': 'vector',
-        'sxc': 'spreadsheet',
-        'sxd': 'vector',
-        'tar': 'archive',
-        'tga': 'image',
-        'tif': 'image',
-        'tiff': 'image',
-        'ttf': 'font',
-        'txt': 'text',
-        'vbs': 'script',
-        'vc': 'spreadsheet',
-        'vml': 'vector',
-        'wav': 'audio',
-        'webp': 'image',
-        'wma': 'audio',
-        'wmv': 'video',
-        'woff': 'font',
-        'xar': 'vector',
-        'xbm': 'image',
-        'xcf': 'image',
-        'xhtml': 'html',
-        'xls': 'spreadsheet',
-        'xlsx': 'spreadsheet',
-        'xml': 'html',
-        'zip': 'archive'
-    }
-
-    file_type_icon = fields.Char('File Type Icon', compute='get_attachment_type', store=True)
-    file_type = fields.Char(related='file_type_icon')     # FIXME remove in trunk
-
-    @api.depends('datas_fname')
-    @api.one
-    def get_attachment_type(self):
-        fileext = os.path.splitext(self.datas_fname or '')[1].lower()[1:]
-        self.file_type_icon = self._fileext_to_type.get(fileext, 'unknown')

--- a/addons/mail/models/mail_message.py
+++ b/addons/mail/models/mail_message.py
@@ -207,12 +207,12 @@ class Message(models.Model):
         partner_tree = dict((partner[0], partner) for partner in partners_names)
 
         # 2. Attachments as SUPERUSER, because could receive msg and attachments for doc uid cannot see
-        attachments_data = attachments.sudo().read(['id', 'datas_fname', 'name', 'file_type_icon'])
+        attachments_data = attachments.sudo().read(['id', 'datas_fname', 'name', 'mimetype'])
         attachments_tree = dict((attachment['id'], {
             'id': attachment['id'],
             'filename': attachment['datas_fname'],
             'name': attachment['name'],
-            'file_type_icon': attachment['file_type_icon'],
+            'mimetype': attachment['mimetype'],
         }) for attachment in attachments_data)
 
         # 3. Tracking values
